### DISCUSSION TITLE
Does 2.4 still have an automated setup option?

### DISCUSSION THREAD

**Discussion by tomjones1977** [Security-Onion-Solutions/securityonion#11314] [2.4] [answered]
### Version

2.4.10

### Installation Method

Other (please provide detail below)

### Description

installation

### Installation Type

other (please provide detail below)

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

4

### RAM

16GB

### Storage for /

100GB

### Storage for /nsm

100GB

### Network Traffic Collection

other (please provide detail below)

### Network Traffic Speeds

Less than 1Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

The older version of setup had an option to create a setup.conf file, documented here:

https://github.com/Security-Onion-Solutions/security-onion/wiki/Automating-Setup

Does this functionality still exist in 2.4?  I can't see it in the docs and did try the -w flag from the older docs but it doesn't work as expected.

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by TOoSmOotH** [comment]
There is no officially supported automated install method in 2.3 or 2.4. If you review the bash code for the install you might be able to automate things via declaring ENV variables.

**Reply by tomjones1977** [reply]
It doesn't look like there's a way to seed variables into the setup script any more.  The install_opt file that is checked for during setup causes the setup to be skipped completely if it exists, even as an empty file.  Short of rewriting the setup script I can't see a way to inject the right data at build time.

**Reply by TOoSmOotH** [reply, marked as answer]
https://stackoverflow.com/a/20909045

Something like that would work.
